Source/Core/Core/AchievementManager.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#ifdef USE_RETRO_ACHIEVEMENTS
+#include <mutex>
+#include <string>
+#include <thread>
+
+#include <rcheevos/include/rc_api_user.h>
+#include <rcheevos/include/rc_runtime.h>
+
+#include "Common/Event.h"
+
+class AchievementManager
+{
+public:
+  static AchievementManager* GetInstance();
+  void Init();
+  void RequestPost();
```

Comment:
From an API design perspective this looks weird to me:

- It's not clear what RequestPost() is or why it's public. If it's a placeholder for something, please clearly document it as such. If it's not a placeholder, why is this something that the user of this object should care about?
- SetPassword isn't really what I'd expect to see -- as far as I understand, the only time the password matters should be when logging in, to exchange for an authnz token. So really the public API I'd expect to see is "Login", not "SetPassword" which implies that the AchievementManager carries the password as mutable state. The password should in fact be mostly a temporary here...
- The current API design doesn't provide any kind of error return path, so there's no way for the API user to know whether the login process happened successfully.

So really what I would be expecting to see here is something akin to:

```
enum class LoginResult { Success, InvalidCredentials, UnknownFailure };
using LoginCallback = std::function<void(LoginResult)>;

// Returns whether the user successfully logged in and we have an RA API token provisioned.
bool IsLoggedIn() const;

LoginResult Login(const std::string& password);  // might block
void LoginAsync(const std::string& password, const LoginCallback& callback);
```

And the implementation should somehow be centered around making that public API work as expected. Login should push some kind of event like `struct LoginEvent { std::string password; LoginCallback callback; };` in a request queue that gets processed by your network thread, etc.